internal/dataapi/server/json.go

```diff
@@ -15,25 +15,48 @@
 package server
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
-	"io"
 
 	"github.com/FerretDB/wire/wirebson"
 	"go.mongodb.org/mongo-driver/v2/bson"
 
 	"github.com/FerretDB/FerretDB/v2/internal/util/lazyerrors"
 )
 
-// marshalJSON encodes wirebson.RawDocument into extended JSON.
-func marshalJSON(raw wirebson.RawDocument, jsonDst io.Writer) error {
-	encoder := bson.NewEncoder(bson.NewExtJSONValueWriter(jsonDst, false, false))
+// marshalSingleJSON takes a single value and marshals it into the JSON format.
+func marshalSingleJSON(v any) (json.RawMessage, error) {
```

Comment:
Does this encode to extended JSON? Does it really take any value? The function signature has changed so we need to document what it takes and what it returns. Could you document those please.